### PR TITLE
Update Azure dependencies

### PR DESCRIPTION
**Pull request by sundowndev**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #1307
| ❓ Documentation  | no 

## Description

This PR upgrades several Azure Go packages. We can't update them individually, because they depend on each other. This update is necessary because we inherit of bugs from older versions (see related issue).

**Note to reviewers**:

- Updated azcore, azidentity, armprivatedns to their latest version
- Several Azure Go module paths were changed. Updated paths for armpostgresql, armstorage, armcompute, armcontainerregistry, armnetwork, armresources. I bumped versions for some of them as well.
- Fixed repositories to fit with the new Azure SDK API. **Battle-tested with my dev environment, please give it a shot too**.

### CODE REVIEW

**Comment by eliecharra** [comment]
> Battle-tested with my dev environment, please give it a shot too.

Does it mean that you have launched all the ACC test suite successfully locally ?

File: go.mod [hunk 1]
```diff
@@ -5,15 +5,15 @@ go 1.16
...
-	github.com/Azure/azure-sdk-for-go/sdk/postgresql/armpostgresql v0.2.0
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute v0.1.0
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/privatedns/armprivatedns v0.1.0
-	github.com/Azure/azure-sdk-for-go/sdk/resources/armresources v0.3.0
-	github.com/Azure/azure-sdk-for-go/sdk/storage/armstorage v0.2.0
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.20.0
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.12.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute v0.2.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerregistry/armcontainerregistry v0.2.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork v0.2.0
```
Comment (eliecharra) [inline go.mod:12]: This one was downgraded from `v0.3.0` to `v0.2.0` is this expected ?

File: go.mod [hunk 2]
```diff
@@ -5,15 +5,15 @@ go 1.16
...
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/postgresql/armpostgresql v0.2.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/privatedns/armprivatedns v0.2.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources v0.2.0
```
Comment (eliecharra) [inline go.mod:15]: `armresources` is also downgraded

**Review by eliecharra — COMMENTED** [review]
Left small comments, otherwise LGTM.

I'll let you merge once you ran flacky test locally to be sure we do not have broken anything.

File: go.mod (hunk 2 above)
Comment (sundowndev) [reply go.mod:15]: Yes, the latest version for the new package is v0.2.0

See https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources

File: go.mod (hunk 1 above)
Comment (sundowndev) [reply go.mod:12]: Yes, the latest version for the new package is v0.2.0

See https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork

**Comment by sundowndev** [comment]
> I'll let you merge once you ran flacky test locally to be sure we do not have broken anything.

@eliecharra Tests successfully passed locally 🎉

**Pull request merged**